src/utils/OpenGLBinding.cpp

```diff
@@ -42,11 +44,18 @@ namespace utils {
             void GetProcAndDevice(nxtProcTable* procs, nxtDevice* device) override {
                 glfwMakeContextCurrent(window);
                 backend::opengl::Init(reinterpret_cast<void*(*)(const char*)>(glfwGetProcAddress), procs, device);
+
+                backendDevice = *device;
+                backend::opengl::InitBackbuffer(backendDevice);
             }
             void SwapBuffers() override {
+                backend::opengl::CommitBackbuffer(backendDevice);
                 glfwSwapBuffers(window);
-                backend::opengl::HACKCLEAR();
+                backend::opengl::HACKCLEAR(backendDevice);
```

Comment:
Ah right, forgot to do this after I moved HACKCLEAR into the device. Unfortunately it doesn't seem to fix the first frame(s) being black.